llvm/include/llvm/Object/DXContainer.h

```diff
@@ -228,6 +273,19 @@ class RootSignature {
       else
         DataSize = sizeof(dxbc::RTS0::v2::RootDescriptor);
       break;
+    case dxbc::RootParameterType::DescriptorTable:
+      uint32_t NumRanges =
+          support::endian::read<uint32_t, llvm::endianness::little>(
+              PartData.begin() + Header.ParameterOffset);
```

Comment:
It's a bit unfortunate that we need to read the data here in order to parse it. Do we do anything elsewhere to make sure that corrupted data doesn't bite us here? That is, does this do something reasonable for an absurdly large incorrect size or does it fall over?

Reply:
The DXContainer reader really needs to be hardened against crashing since this code may be used in drivers and/or runtime libraries.

We should at least validate that the offset is within the part's range and error appropriately.